include/edac.h

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief EDAC API header file
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_EDAC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_EDAC_H_
+
+#include <sys/types.h>
+
+typedef int (*edac_api_inject_addr_set)(const struct device *dev,
+					uint64_t addr);
+typedef uint64_t (*edac_api_inject_addr_get)(const struct device *dev);
+
+typedef int (*edac_api_inject_addr_mask_set)(const struct device *dev,
+					     uint64_t mask);
+typedef uint64_t (*edac_api_inject_addr_mask_get)(const struct device *dev);
+
+typedef int (*edac_api_inject_ctrl_set)(const struct device *dev, uint32_t ctrl);
+
+typedef uint64_t (*edac_api_ecc_error_log_get)(const struct device *dev);
+typedef void (*edac_api_ecc_error_log_clear)(const struct device *dev);
+
+typedef uint64_t (*edac_api_parity_error_log_get)(const struct device *dev);
+typedef void (*edac_api_parity_error_log_clear)(const struct device *dev);
+
+typedef unsigned int (*edac_api_errors_cor_get)(const struct device *dev);
+typedef unsigned int (*edac_api_errors_uc_get)(const struct device *dev);
+
+typedef void (*edac_notify_callback_t)(const struct device *dev,
+				       void *data);
+typedef int (*edac_api_notify_cb_set)(const struct device *dev,
+				      edac_notify_callback_t cb);
```

Comment:
Should we follow some naming convention for the function typedef, Tomasz used `_f` prefix for ivshmem PR, should we follow suit?

Reply:
added `_f` suffix.